File: www/PluginTemplate.js

var exec = require('cordova/exec');

var PLUGIN_NAME = 'PluginTemplate';

var PluginTemplate = {
  sayHello: function(words, cb) {
    exec(cb, null, PLUGIN_NAME, 'sayHello', [words]);
  }
};

module.exports = PluginTemplate;

File: src/ios/PluginTemplate.m
#import "PluginTemplate.h"

#import <Cordova/CDVAvailability.h>

@implementation PluginTemplate

- (void)pluginInitialize {
    //add observer or initialize here
}

- (void)sayHello:(CDVInvokedUrlCommand *)command {

    NSString* words = [command.arguments objectAtIndex:0];
    NSLog(@"*******js says: %@", words);

    NSString* echo = @"I am native!";
    CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:echo];
    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
}

@end
